pkg/k8s/client.go

```diff
@@ -89,3 +94,33 @@ func CreateTPRClient(endpoint, kubeCfgPath string) (*rest.RESTClient, error) {
 
 	return rest.RESTClientFor(config)
 }
+
+// AnnotateNodeCIDR writes both v4 and v6 CIDRs in the given k8sNode.
+// In case of failure while updating the node, this function while spawn a go
+// routine to retry the node update indefinitely.
+func AnnotateNodeCIDR(c kubernetes.Interface, k8sNode *v1.Node, v4CIDR, v6CIDR *net.IPNet) {
+	// register IP CIDRs in node's annotations
+	log.Debugf("k8s: Storing IPv4 CIDR %s in k8s node %s's annotations", v4CIDR, k8sNode.Name)
+	log.Debugf("k8s: Storing IPv6 CIDR %s in k8s node %s's annotations", v6CIDR, k8sNode.Name)
+	k8sNode.Annotations[Annotationv4CIDRName] = v4CIDR.String()
+	k8sNode.Annotations[Annotationv6CIDRName] = v6CIDR.String()
+
+	_, err := c.CoreV1().Nodes().Update(k8sNode)
+	if err != nil {
+		go func(c kubernetes.Interface, k8sServerNode *v1.Node, v4CIDR, v6CIDR *net.IPNet, err error) {
+			// TODO: Retry forever?
+			for n := 0; err != nil; {
+				log.Errorf("K8s: unable to update node %s with IPv6 CIDR annotation: %s, retrying...", k8sServerNode.Name, err)
+				// In case of an error let's retry until
+				// we were able to set the annotations properly
+				k8sServerNode.Annotations[Annotationv4CIDRName] = v4CIDR.String()
+				k8sServerNode.Annotations[Annotationv6CIDRName] = v6CIDR.String()
+				k8sServerNode, err = c.CoreV1().Nodes().Update(k8sNode)
```

Comment:
I see, that's good then. I think we should have this running in the background and re-add the annotation in case someone removes the annotation for some reason. We could only trigger the re-add through an agent restart.